diff --git a/tmp/_oc-completions b/tmp/_oc-completions-patched
index 9c7d488..9b6f1b1 100644
--- a/tmp/_oc-completions
+++ b/tmp/_oc-completions-patched
@@ -1,5 +1,65 @@
 
-# bash completion for oc                                   -*- shell-script -*-
+_oc_af-arbitrary-command()
+{
+    last_command="oc_af_arbitrary_command"
+    flags=()
+    two_word_flags=()
+    local_nonpersistent_flags=()
+    flags_with_completion=()
+    flags_completion=()
+
+    flags+=("-h")
+    flags+=("-A")
+    flags+=("-n")
+    flags+=("-c")
+
+    two_word_flags+=("-n")
+    flags_with_completion+=("-n")
+    flags_completion+=("_watch-namespace_completions")
+}
+
+_oc_restart-af-services()
+{
+    last_command="oc_restart_af_services"
+    flags=()
+    two_word_flags=()
+    local_nonpersistent_flags=()
+    flags_with_completion=()
+    flags_completion=()
+
+    flags+=("-h")
+    flags+=("--minio")
+    flags+=("--artifactory")
+    flags+=("--no-dry-run")
+    flags+=("--no-primary")
+    flags+=("-A")
+
+    two_word_flags+=("-n")
+    flags_with_completion+=("-n")
+    flags_completion+=("_watch-namespace_completions")
+}
+
+_oc_watch-namespace()
+{
+    last_command="oc_watch_namespace"
+    flags=()
+    two_word_flags=()
+    local_nonpersistent_flags=()
+    flags_with_completion=()
+    flags_completion=()
+
+    flags+=("-h")
+    # flags+=("-r")
+    two_word_flags+=("-r")
+    flags_with_completion+=("-r")
+    flags_completion+=("_watch-namespace_completions")
+    # flags_completion+=("__oc_region_complete")
+
+    two_word_flags+=("-n")
+    flags_with_completion+=("-n")
+    flags_completion+=("_watch-namespace_completions")
+
+}
 
 __oc_debug()
 {
@@ -297,13 +357,8 @@ __oc_parse_get()
     fi
 }
 
-__oc_get_namespaces()
-{
-    local template oc_out
-    template="{{ range .items  }}{{ .metadata.name }} {{ end }}"
-    if oc_out=$(oc get -o template --template="${template}" namespace 2>/dev/null); then
-        COMPREPLY=( $( compgen -W "${oc_out[*]}" -- "$cur" ) )
-    fi
+__oc_get_namespaces () {
+    _watch-namespace_completions
 }
 
 __oc_get_resource()
@@ -20474,13 +20529,34 @@ _oc_whoami()
     noun_aliases=()
 }
 
-_oc_root_command()
-{
+_oc_root_command () {
     last_command="oc"
 
     command_aliases=()
 
     commands=()
+    commands+=("fields")
+    commands+=("images")
+    commands+=("modify-secret")
+    commands+=("neat")
+    commands+=("node-shell")
+    commands+=("rbac-lookup")
+    commands+=("sick-pods")
+    commands+=("tmux-exec")
+    commands+=("topology")
+    commands+=("ttsum")
+    commands+=("view-secret")
+    commands+=("warp")
+    commands+=("who-can")
+    commands+=("get-all-resources")
+    commands+=("watch-namespace")
+    commands+=("tmux-exec")
+    commands+=("velero-annotate-all-volumes-for-pod")
+    commands+=("delete-namespace-finalizer")
+    commands+=("non-running-pods")
+    commands+=("non-running-nodes")
+    commands+=("restart-af-services")
+    commands+=("af-arbitrary-command")
     commands+=("adm")
     commands+=("annotate")
     commands+=("api-resources")
@@ -20637,3 +20713,4 @@ else
 fi
 
 # ex: ts=4 sw=4 et filetype=sh
+
